hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/AbstractClientInternalCacheProxy.java

```diff
@@ -240,7 +240,9 @@ protected ClientInvocationFuture invoke(ClientMessage req, Data keyData, int com
         validateConfiguredTypes(cacheConfig, key);
 
         Data keyData = toData(key);
-        return getAndRemoveInternal(keyData, true);
+        ClientDelegatingFuture<T> delegatingFuture = getAndRemoveInternal(keyData, true);
+        onGetAndRemoveAsyncInternal(key, keyData, delegatingFuture, null);
```

Comment:
This is the missing method call in the client cache proxy on a sync `getAndRemove()` call. The `onGetAndRemoveAsyncInternal()` is overridden in the Near Cache proxy, to handle the local invalidations.